examples/pytorch/rgcn/model.py

```diff
@@ -46,3 +47,93 @@ def forward(self, g, h, r, norm):
         for layer in self.layers:
             h = layer(g, h, r, norm)
         return h
+
+class RelGraphEmbedLayer(nn.Module):
+    r"""Embedding layer for featureless heterograph.
+    Parameters
+    ----------
+    dev_id : int
+        Device to run the layer.
+    num_nodes : int
+        Number of nodes.
+    node_tides : tensor
+        Storing the node type id for each node starting from 0
+    num_of_ntype : int
+        Number of node types
+    input_size : list of int
+        A list of input feature size for each node type. If None, we then 
+        treat certain input feature as an one-hot encoding feature.
+    embed_size : int
+        Output embed size
+    embed_name : str, optional
+        Embed name
+    """
+    def __init__(self,
+                 dev_id,
+                 num_nodes,
+                 node_tids,
+                 num_of_ntype,
+                 input_size,
+                 embed_size,
+                 embed_name='embed'):
+        super(RelGraphEmbedLayer, self).__init__()
+        self.dev_id = dev_id
+        self.embed_size = embed_size
+        self.embed_name = embed_name
+        self.num_nodes = num_nodes
+
+        # create weight embeddings for each node for each relation
+        self.embeds = nn.ParameterDict()
+        self.num_of_ntype = num_of_ntype
+        self.idmap = th.empty(num_nodes).long()
+
+        for ntype in range(num_of_ntype):
+            if input_size[ntype] is not None:
+                loc = node_tids == ntype
+                input_emb_size = node_tids[loc].shape[0]
+                embed = nn.Parameter(th.Tensor(input_emb_size, self.embed_size))
+                nn.init.xavier_uniform_(embed, gain=nn.init.calculate_gain('relu'))
+                self.embeds[str(ntype)] = embed
+            else:
+                loc = node_tids == ntype
+                num_subnodes = node_tids[loc].shape[0]
+                self.idmap[loc] = th.arange(num_subnodes)
+                none_embed = nn.Parameter(th.Tensor(num_subnodes, self.embed_size))
+                nn.init.xavier_uniform_(none_embed, gain=nn.init.calculate_gain('relu'))
+                self.embeds[str(ntype)] = none_embed
```

Comment:
what is the difference between these two branches? and why do you want to have nodes of different types in different embedding tensors?

Reply:
This is for nodes having initial embeddings.
I want to write another PR to include the example.